roles/openshift_master/tasks/main.yml

```diff
@@ -283,6 +284,20 @@
   when: openshift_master_ha | bool and openshift.master.cluster_method == 'native'
   register: start_result
 
+# A separate wait is required here for native HA since notifies will
+# be resolved after all tasks in the role.
+- name: Wait for API to become available
+  # Using curl here since the uri module requires python-httplib2 and
+  # wait_for port doesn't provide health information.
+  command: >
+    curl -k --head --silent {{ openshift.master.api_url }}
+  register: api_available_output
+  until: api_available_output.stdout.find("200 OK") != -1
+  retries: 120
+  delay: 1
+  changed_when: false
+  when: openshift_master_ha | bool and openshift.master.cluster_method == 'native'
```

Comment:
Why not move this down below the set_fact and make it conditional on start_result?